pycbc/io/live.py

```diff
@@ -332,6 +369,21 @@ def upload(self, fname, gracedb_server=None, testing=True,
             pylab.xlabel('Frequency (Hz)')
             pylab.ylabel('ASD')
             pylab.savefig(psd_series_plot_fname)
+            pylab.close()
+
+        if self.probabilities is not None:
+            prob_fname = fname.replace('.xml.gz', '_probs.json')
+            prob_plot_fname = prob_fname.replace('.json', '.png')
+            probabilities = json.loads(self.probabilities)
```

Comment:
Related to another comment above, I would not store the probabilities in JSON and then un-serialize them back to a dict. You can just store the dict and convert it to JSON when you actually write the JSON file.